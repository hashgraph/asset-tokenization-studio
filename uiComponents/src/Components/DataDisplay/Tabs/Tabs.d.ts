import type { tabsAnatomy as ChakraTabsAnatomy } from "@chakra-ui/anatomy";
import type { ComponentWithAs } from "@chakra-ui/system";
import type { TabListProps as ChakraTabListProps, TabPanelsProps as ChakraTabPanelsProps, TabProps as ChakraTabProps, TabsProps as ChakraTabsProps } from "@chakra-ui/tabs";
import type { ReactNode } from "react";
import type { BaseMultiStyleConfiguration } from "@Theme/Components/BaseMultiStyleConfiguration";
export declare const tabsPartsList: typeof ChakraTabsAnatomy.keys;
type Parts = typeof tabsPartsList;
export interface TabsThemeConfiguration extends BaseMultiStyleConfiguration<Parts> {
}
export interface TabProps extends Omit<ChakraTabProps, "children"> {
    header: string | ReactNode;
    content: string | ReactNode;
}
export interface TabsProps extends Omit<ChakraTabsProps, "children"> {
    tabs: TabProps[];
    listProps?: ChakraTabListProps;
    panelsProps?: ChakraTabPanelsProps;
    variant?: string;
}
export declare const Tabs: ComponentWithAs<"div", TabsProps>;
export {};
