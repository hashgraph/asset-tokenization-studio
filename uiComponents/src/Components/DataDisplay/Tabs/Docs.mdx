import {
  Story,
  Meta,
  Title,
  ArgTypes,
  Primary,
  Stories,
  Source,
} from "@storybook/addon-docs";
import { Grid, GridItem, Text, Link, Center, Flex } from "@chakra-ui/react";
import * as TabsStories from "./Tabs.stories";

<Meta of={TabsStories} />

<Title />

<a
  href="https://www.figma.com/file/YF8AvIq0MclfhGDvKgzypp/ioBricks-Design-System?type=design&node-id=2922%3A35786&mode=design&t=k4jSCm0Q182mRyVn-1"
  target="\_blank"
>
  View figma design
</a>

<Primary />

## How to set up

### Config theme

Add the following configurations to the `theme.tsx` and `Tabs.ts` files, respectively.

```tsx dark
const theme = {
  components: {
    Tabs: ConfigTabs,
  },
};
```

```tsx dark
const ConfigTabs: TabsThemeConfiguration = {
  parts: tabsPartsList,
  baseStyle: {},
};
```

### Component parts

You have these parts available: `"root" | "tab" | "tablist" | "tabpanel" | "tabpanels" | "indicator"` (`tabsPartsList`)

## Props

<ArgTypes of={TabsStories} />

## Stories

<Stories title="" />
