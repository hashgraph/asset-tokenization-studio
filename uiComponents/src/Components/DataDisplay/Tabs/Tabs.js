var __assign = (this && this.__assign) || function () {
    __assign = Object.assign || function(t) {
        for (var s, i = 1, n = arguments.length; i < n; i++) {
            s = arguments[i];
            for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p))
                t[p] = s[p];
        }
        return t;
    };
    return __assign.apply(this, arguments);
};
var __rest = (this && this.__rest) || function (s, e) {
    var t = {};
    for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p) && e.indexOf(p) < 0)
        t[p] = s[p];
    if (s != null && typeof Object.getOwnPropertySymbols === "function")
        for (var i = 0, p = Object.getOwnPropertySymbols(s); i < p.length; i++) {
            if (e.indexOf(p[i]) < 0 && Object.prototype.propertyIsEnumerable.call(s, p[i]))
                t[p[i]] = s[p[i]];
        }
    return t;
};
import React, { useMemo } from "react";
import { forwardRef } from "@chakra-ui/system";
import { useMultiStyleConfig as useChakraMultiStyleConfig } from "@chakra-ui/react";
import { Tab as ChakraTab, TabList as ChakraTabList, TabPanel as ChakraTabPanel, TabPanels as ChakraTabPanels, Tabs as ChakraTabs, TabIndicator as ChakraTabIndicator, } from "@chakra-ui/tabs";
export var tabsPartsList = [
    "root",
    "tab",
    "tablist",
    "tabpanel",
    "tabpanels",
    "indicator",
];
export var Tabs = forwardRef(function (_a, ref) {
    var tabs = _a.tabs, listProps = _a.listProps, panelsProps = _a.panelsProps, variant = _a.variant, isFittedArg = _a.isFitted, props = __rest(_a, ["tabs", "listProps", "panelsProps", "variant", "isFitted"]);
    var isFitted = useMemo(function () {
        if (isFittedArg)
            return true;
        if (variant === "table")
            return true;
    }, [isFittedArg, variant]);
    var styles = useChakraMultiStyleConfig("Tabs", { variant: variant, isFitted: isFitted });
    return (React.createElement(ChakraTabs, __assign({ ref: ref, sx: styles.root }, props),
        React.createElement(ChakraTabList, __assign({ sx: styles.tablist }, listProps), tabs.map(function (_a, ix) {
            var content = _a.content, header = _a.header, tab = __rest(_a, ["content", "header"]);
            return (React.createElement(ChakraTab, __assign({ key: "tab_title_".concat(ix), sx: styles.tab }, tab), header));
        })),
        React.createElement(ChakraTabIndicator, { sx: styles.indicator }),
        React.createElement(ChakraTabPanels, __assign({ sx: styles.tabpanels }, panelsProps), tabs.map(function (_a, ix) {
            var content = _a.content;
            return (React.createElement(ChakraTabPanel, { sx: styles.tabpanel, key: "tab_content_".concat(ix) }, content));
        }))));
});
