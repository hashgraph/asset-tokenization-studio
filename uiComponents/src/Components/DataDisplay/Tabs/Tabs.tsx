import React, { useMemo } from "react";
import type { tabsAnatomy as ChakraTabsAnatomy } from "@chakra-ui/anatomy";

import type { ComponentWithAs } from "@chakra-ui/system";
import { forwardRef } from "@chakra-ui/system";
import { useMultiStyleConfig as useChakraMultiStyleConfig } from "@chakra-ui/react";
import type {
  TabListProps as ChakraTabListProps,
  TabPanelsProps as ChakraTabPanelsProps,
  TabProps as ChakraTabProps,
  TabsProps as ChakraTabsProps,
} from "@chakra-ui/tabs";
import {
  Tab as ChakraTab,
  TabList as ChakraTabList,
  TabPanel as ChakraTabPanel,
  TabPanels as ChakraTabPanels,
  Tabs as ChakraTabs,
  TabIndicator as ChakraTabIndicator,
} from "@chakra-ui/tabs";
import type { ReactNode } from "react";
import type { BaseMultiStyleConfiguration } from "@Theme/Components/BaseMultiStyleConfiguration";

export const tabsPartsList: typeof ChakraTabsAnatomy.keys = [
  "root",
  "tab",
  "tablist",
  "tabpanel",
  "tabpanels",
  "indicator",
];

type Parts = typeof tabsPartsList;

export interface TabsThemeConfiguration
  extends BaseMultiStyleConfiguration<Parts> {}
export interface TabProps extends Omit<ChakraTabProps, "children"> {
  header: string | ReactNode;
  content: string | ReactNode;
}
export interface TabsProps extends Omit<ChakraTabsProps, "children"> {
  tabs: TabProps[];
  listProps?: ChakraTabListProps;
  panelsProps?: ChakraTabPanelsProps;
  variant?: string;
}
export const Tabs: ComponentWithAs<"div", TabsProps> = forwardRef<
  TabsProps,
  "div"
>(
  (
    {
      tabs,
      listProps,
      panelsProps,
      variant,
      isFitted: isFittedArg,
      ...props
    }: TabsProps,
    ref
  ) => {
    const isFitted = useMemo(() => {
      if (isFittedArg) return true;
      if (variant === "table") return true;
    }, [isFittedArg, variant]);
    const styles = useChakraMultiStyleConfig("Tabs", { variant, isFitted });
    return (
      <ChakraTabs ref={ref} sx={styles.root} {...props}>
        <ChakraTabList sx={styles.tablist} {...listProps}>
          {tabs.map(({ content, header, ...tab }, ix) => (
            <ChakraTab key={`tab_title_${ix}`} sx={styles.tab} {...tab}>
              {header}
            </ChakraTab>
          ))}
        </ChakraTabList>

        <ChakraTabIndicator sx={styles.indicator} />

        <ChakraTabPanels sx={styles.tabpanels} {...panelsProps}>
          {tabs.map(({ content }, ix) => (
            <ChakraTabPanel sx={styles.tabpanel} key={`tab_content_${ix}`}>
              {content}
            </ChakraTabPanel>
          ))}
        </ChakraTabPanels>
      </ChakraTabs>
    );
  }
);
