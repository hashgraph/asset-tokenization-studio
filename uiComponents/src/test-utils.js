import { ChakraProvider, extendTheme } from "@chakra-ui/react";
import { render as testingLibraryRender } from "@testing-library/react";
import React from "react";
import { BasePlatformTheme } from "./Theme/theme";
var DefaultTheme = extendTheme(BasePlatformTheme);
export var render = function (ui, theme) {
    if (theme === void 0) { theme = DefaultTheme; }
    var wrapper = function (_a) {
        var children = _a.children;
        return (React.createElement(ChakraProvider, { theme: theme }, children));
    };
    return testingLibraryRender(ui, { wrapper: wrapper });
};
