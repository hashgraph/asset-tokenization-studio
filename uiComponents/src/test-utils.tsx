import { ChakraProvider, extendTheme } from "@chakra-ui/react";
import type { RenderResult } from "@testing-library/react";
import { render as testingLibraryRender } from "@testing-library/react";
import React from "react";
import { BasePlatformTheme } from "./Theme/theme";

const DefaultTheme = extendTheme(BasePlatformTheme);

export const render = (ui: JSX.Element, theme = DefaultTheme): RenderResult => {
  const wrapper = ({ children }: { children?: React.ReactNode }) => (
    <ChakraProvider theme={theme}>{children}</ChakraProvider>
  );

  return testingLibraryRender(ui, { wrapper });
};
