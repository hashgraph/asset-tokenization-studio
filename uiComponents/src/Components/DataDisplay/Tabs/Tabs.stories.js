var __assign = (this && this.__assign) || function () {
    __assign = Object.assign || function(t) {
        for (var s, i = 1, n = arguments.length; i < n; i++) {
            s = arguments[i];
            for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p))
                t[p] = s[p];
        }
        return t;
    };
    return __assign.apply(this, arguments);
};
var __spreadArray = (this && this.__spreadArray) || function (to, from, pack) {
    if (pack || arguments.length === 2) for (var i = 0, l = from.length, ar; i < l; i++) {
        if (ar || !(i in from)) {
            if (!ar) ar = Array.prototype.slice.call(from, 0, i);
            ar[i] = from[i];
        }
    }
    return to.concat(ar || Array.prototype.slice.call(from));
};
import React from "react";
import { Tabs } from "./Tabs";
var tabs = __spreadArray([], Array(5), true).map(function (_el, index) { return ({
    header: "Tab ".concat(index + 1),
    content: "Tab ".concat(index + 1, " Content ")
}); });
var meta = {
    title: "Design System/Data Display/Tabs",
    component: Tabs,
    argTypes: {
        tabs: {
            description: "Tabs object list.",
            table: {
                type: {
                    summary: "TabProps[]",
                    detail: "\n            TabProps extends ChakraTabProps {\n            header: string | ReactNode;\n            content: string | ReactNode;\n          }"
                }
            },
            control: {
                type: "object"
            }
        }
    },
    parameters: {
        design: {
            type: "figma",
            url: "https://www.figma.com/file/yn0TWBlB8x5WbsEY8BdML7/Palm-Design-System?node-id=1353%3A7922"
        }
    },
    args: {
        tabs: tabs
    }
};
export default meta;
var Template = function (args) { return React.createElement(Tabs, __assign({}, args)); };
export var Default = Template.bind({});
Default.args = {};
export var TableTabs = Template.bind({});
TableTabs.args = {
    variant: "table"
};
