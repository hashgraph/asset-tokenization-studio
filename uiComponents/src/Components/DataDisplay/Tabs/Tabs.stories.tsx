import type { Meta, StoryFn } from "@storybook/react";
import React from "react";

import { Tabs } from "./Tabs";

const tabs = [...Array(5)].map((_el, index) => ({
  header: `Tab ${index + 1}`,
  content: `Tab ${index + 1} Content `,
}));

const meta = {
  title: "Design System/Data Display/Tabs",
  component: Tabs,
  argTypes: {
    tabs: {
      description: "Tabs object list.",
      table: {
        type: {
          summary: `TabProps[]`,
          detail: `
            TabProps extends ChakraTabProps {
            header: string | ReactNode;
            content: string | ReactNode;
          }`,
        },
      },
      control: {
        type: "object",
      },
    },
  },
  parameters: {
    design: {
      type: "figma",
      url: "https://www.figma.com/file/yn0TWBlB8x5WbsEY8BdML7/Palm-Design-System?node-id=1353%3A7922",
    },
  },
  args: {
    tabs,
  },
} as Meta<typeof Tabs>;
export default meta;

const Template: StoryFn<typeof Tabs> = (args) => <Tabs {...args} />;

export const Default = Template.bind({});
Default.args = {};

export const TableTabs = Template.bind({});
TableTabs.args = {
  variant: "table",
};
