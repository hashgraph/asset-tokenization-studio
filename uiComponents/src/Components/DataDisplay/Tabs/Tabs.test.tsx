import { render } from "@/test-utils";
import React from "react";
import type { TabsProps } from "./Tabs";
import { Tabs } from "./Tabs";

const tabs = [...Array(5)].map((_el, index) => ({
  header: `Tab ${index + 1}`,
  content: `Tab ${index + 1} Content `,
  "data-testid": `tab-header-${index + 1}`,
}));

describe(`<Tabs />`, () => {
  const factoryComponent = (props: TabsProps) => render(<Tabs {...props} />);

  test("renders correctly", () => {
    const component = factoryComponent({ tabs });
    expect(component.asFragment()).toMatchSnapshot();
  });

  test("Renders the number of tabs passed by arg", () => {
    const component = factoryComponent({ tabs });
    expect(component.getAllByTestId(/tab-header.*/).length).toBe(5);
    expect(component.asFragment()).toMatchSnapshot();
  });
});
