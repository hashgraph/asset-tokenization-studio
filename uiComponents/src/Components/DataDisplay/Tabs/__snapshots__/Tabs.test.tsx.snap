// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Tabs /> Renders the number of tabs passed by arg 1`] = `
<DocumentFragment>
  <div
    class="chakra-tabs css-1bx5ylf"
  >
    <div
      aria-orientation="horizontal"
      class="chakra-tabs__tablist css-3o7rpu"
      role="tablist"
    >
      <button
        aria-controls="tabs-:r1:--tabpanel-0"
        aria-selected="true"
        class="chakra-tabs__tab css-8mj2xm"
        data-index="0"
        data-testid="tab-header-1"
        id="tabs-:r1:--tab-0"
        role="tab"
        tabindex="0"
        type="button"
      >
        Tab 1
      </button>
      <button
        aria-controls="tabs-:r1:--tabpanel-1"
        aria-selected="false"
        class="chakra-tabs__tab css-8mj2xm"
        data-index="1"
        data-testid="tab-header-2"
        id="tabs-:r1:--tab-1"
        role="tab"
        tabindex="-1"
        type="button"
      >
        Tab 2
      </button>
      <button
        aria-controls="tabs-:r1:--tabpanel-2"
        aria-selected="false"
        class="chakra-tabs__tab css-8mj2xm"
        data-index="2"
        data-testid="tab-header-3"
        id="tabs-:r1:--tab-2"
        role="tab"
        tabindex="-1"
        type="button"
      >
        Tab 3
      </button>
      <button
        aria-controls="tabs-:r1:--tabpanel-3"
        aria-selected="false"
        class="chakra-tabs__tab css-8mj2xm"
        data-index="3"
        data-testid="tab-header-4"
        id="tabs-:r1:--tab-3"
        role="tab"
        tabindex="-1"
        type="button"
      >
        Tab 4
      </button>
      <button
        aria-controls="tabs-:r1:--tabpanel-4"
        aria-selected="false"
        class="chakra-tabs__tab css-8mj2xm"
        data-index="4"
        data-testid="tab-header-5"
        id="tabs-:r1:--tab-4"
        role="tab"
        tabindex="-1"
        type="button"
      >
        Tab 5
      </button>
    </div>
    <div
      class="chakra-tabs__tab-indicator css-1axjpr8"
      style="position: absolute; transition-property: left, right, top, bottom, height, width; transition-duration: 0ms; transition-timing-function: cubic-bezier(0, 0, 0.2, 1); left: 0px; width: 0px;"
    />
    <div
      class="chakra-tabs__tab-panels css-8atqhb"
    >
      <div
        aria-labelledby="tabs-:r1:--tab-0"
        class="chakra-tabs__tab-panel css-1752b0c"
        id="tabs-:r1:--tabpanel-0"
        role="tabpanel"
        tabindex="0"
      >
        Tab 1 Content 
      </div>
      <div
        aria-labelledby="tabs-:r1:--tab-1"
        class="chakra-tabs__tab-panel css-1752b0c"
        hidden=""
        id="tabs-:r1:--tabpanel-1"
        role="tabpanel"
        tabindex="0"
      >
        Tab 2 Content 
      </div>
      <div
        aria-labelledby="tabs-:r1:--tab-2"
        class="chakra-tabs__tab-panel css-1752b0c"
        hidden=""
        id="tabs-:r1:--tabpanel-2"
        role="tabpanel"
        tabindex="0"
      >
        Tab 3 Content 
      </div>
      <div
        aria-labelledby="tabs-:r1:--tab-3"
        class="chakra-tabs__tab-panel css-1752b0c"
        hidden=""
        id="tabs-:r1:--tabpanel-3"
        role="tabpanel"
        tabindex="0"
      >
        Tab 4 Content 
      </div>
      <div
        aria-labelledby="tabs-:r1:--tab-4"
        class="chakra-tabs__tab-panel css-1752b0c"
        hidden=""
        id="tabs-:r1:--tabpanel-4"
        role="tabpanel"
        tabindex="0"
      >
        Tab 5 Content 
      </div>
    </div>
  </div>
  <span
    hidden=""
    id="__chakra_env"
  />
</DocumentFragment>
`;

exports[`<Tabs /> renders correctly 1`] = `
<DocumentFragment>
  <div
    class="chakra-tabs css-1bx5ylf"
  >
    <div
      aria-orientation="horizontal"
      class="chakra-tabs__tablist css-3o7rpu"
      role="tablist"
    >
      <button
        aria-controls="tabs-:r0:--tabpanel-0"
        aria-selected="true"
        class="chakra-tabs__tab css-8mj2xm"
        data-index="0"
        data-testid="tab-header-1"
        id="tabs-:r0:--tab-0"
        role="tab"
        tabindex="0"
        type="button"
      >
        Tab 1
      </button>
      <button
        aria-controls="tabs-:r0:--tabpanel-1"
        aria-selected="false"
        class="chakra-tabs__tab css-8mj2xm"
        data-index="1"
        data-testid="tab-header-2"
        id="tabs-:r0:--tab-1"
        role="tab"
        tabindex="-1"
        type="button"
      >
        Tab 2
      </button>
      <button
        aria-controls="tabs-:r0:--tabpanel-2"
        aria-selected="false"
        class="chakra-tabs__tab css-8mj2xm"
        data-index="2"
        data-testid="tab-header-3"
        id="tabs-:r0:--tab-2"
        role="tab"
        tabindex="-1"
        type="button"
      >
        Tab 3
      </button>
      <button
        aria-controls="tabs-:r0:--tabpanel-3"
        aria-selected="false"
        class="chakra-tabs__tab css-8mj2xm"
        data-index="3"
        data-testid="tab-header-4"
        id="tabs-:r0:--tab-3"
        role="tab"
        tabindex="-1"
        type="button"
      >
        Tab 4
      </button>
      <button
        aria-controls="tabs-:r0:--tabpanel-4"
        aria-selected="false"
        class="chakra-tabs__tab css-8mj2xm"
        data-index="4"
        data-testid="tab-header-5"
        id="tabs-:r0:--tab-4"
        role="tab"
        tabindex="-1"
        type="button"
      >
        Tab 5
      </button>
    </div>
    <div
      class="chakra-tabs__tab-indicator css-1axjpr8"
      style="position: absolute; transition-property: left, right, top, bottom, height, width; transition-duration: 0ms; transition-timing-function: cubic-bezier(0, 0, 0.2, 1); left: 0px; width: 0px;"
    />
    <div
      class="chakra-tabs__tab-panels css-8atqhb"
    >
      <div
        aria-labelledby="tabs-:r0:--tab-0"
        class="chakra-tabs__tab-panel css-1752b0c"
        id="tabs-:r0:--tabpanel-0"
        role="tabpanel"
        tabindex="0"
      >
        Tab 1 Content 
      </div>
      <div
        aria-labelledby="tabs-:r0:--tab-1"
        class="chakra-tabs__tab-panel css-1752b0c"
        hidden=""
        id="tabs-:r0:--tabpanel-1"
        role="tabpanel"
        tabindex="0"
      >
        Tab 2 Content 
      </div>
      <div
        aria-labelledby="tabs-:r0:--tab-2"
        class="chakra-tabs__tab-panel css-1752b0c"
        hidden=""
        id="tabs-:r0:--tabpanel-2"
        role="tabpanel"
        tabindex="0"
      >
        Tab 3 Content 
      </div>
      <div
        aria-labelledby="tabs-:r0:--tab-3"
        class="chakra-tabs__tab-panel css-1752b0c"
        hidden=""
        id="tabs-:r0:--tabpanel-3"
        role="tabpanel"
        tabindex="0"
      >
        Tab 4 Content 
      </div>
      <div
        aria-labelledby="tabs-:r0:--tab-4"
        class="chakra-tabs__tab-panel css-1752b0c"
        hidden=""
        id="tabs-:r0:--tabpanel-4"
        role="tabpanel"
        tabindex="0"
      >
        Tab 5 Content 
      </div>
    </div>
  </div>
  <span
    hidden=""
    id="__chakra_env"
  />
</DocumentFragment>
`;
